docs/src/ref/tectonic-toml.md

```diff
@@ -24,101 +24,52 @@ pubish = false
-
-### `output.shell_escape_cwd`
-
-The working directory path to use for “shell escape”. The default is a
-temporary directory if `output.shell_escape` is true, else it's disabled.
-The path can be absolute or relative to the root file, but it must exist.
-Specifying this path automatically sets `output.shell_escape` to true.
-
-### `output.preamble`
-
-The preamble file to build the document with for this output. This defaults to
-`"_preamble.tex"` within the `src` directory. Typically this file will contain
-document setup steps.
-
-### `output.index`
-
-The index file to build the document with for this output. This defaults to
-`"index.tex"` within the `src` directory. Typically this file will contain
-the body of the document.
-
-### `output.postamble`
-
-The postamble file to build the document with for this output. This defaults to
-`"_postamble.tex"` within the `src` directory. Typically this file will contain
-document closing steps.
+# One (of possibly many) output specifications.
+[[output]]
+
+# This output's name. By default, build products for each output will be
+# placed in the build directory under subdirectory with this name.
+name = "output name"
+
+# The output's type. Right now, only "pdf" is valid.
+type = "pdf"
+
+# The TeX "format" of preloaded macros to use when compiling the document.
+# This is optional, with a default of "latex" (which corresponds to the
+# standard LaTeX format). The exact set of formats that are supported will
+# depend on the bundle that is being used.
+tex_format = "latex"
+
+# Whether the TeX “shell escape”, AKA `\write18`, mechanism is allowed.
+# This is optional and defaults to false.
+#
+# Shell-escape is insecure, since it give the document access to your shell.
+# It also is non-portable, because it requires your document to be built
+# is run in an environment where a shell exists.
+# Naturally, its use is strongly discouraged, but some packages depend on
+# this feature.
+shell_escape = false
+
+# The working directory path to use for “shell escape”. The default is a
+# temporary directory if `output.shell_escape` is true, else it's disabled.
+# The path can be absolute or relative to the root file, but it must exist.
+# Specifying this path automatically sets `output.shell_escape` to true.
+# This is optional, and defaults to a temporary directory.
+shell_escape_cwd = "string"
+
+# The input files we'll use to build this document.
+# Usually, this is a path relative to the `./src` directory.
+#
+# This may also be an array of paths, which are concatenated
+# while building. You could, for example, define:
+# inputs = ["preamble.tex", "main.tex"]
+#
+# Finally, you may include an "inline" document as follows:
+# inputs = [
+#   { inline = "\\documentclass[a4paper]{article}" },
+#   "main.tex"
+# ]
+# This will insert "\documentclass[a4paper]{article}" before main.tex
+# (with a newline), allowing you to set options without making a new file.
+inputs = "main.tex"
+```
```

Comment:
This new documentation structure is an improvement, I think. But the reference documentation should document the `preamble`, etc., fields, even if they're deprecated.